client.go

```diff
@@ -178,6 +179,28 @@ func (c *Client) ExchangeContext(ctx context.Context, m *Msg, a string) (
 	return r, rtt, nil
 }
 
+// Client.DialTimeout acts like DialTimeout, but uses the client's configuration
+func (c *Client) DialTimeoutFunc(network, address string) (conn *Conn, err error) {
+	dialer := net.Dialer{Timeout: c.dialTimeout(), LocalAddr: c.LocalAddr}
+
+	conn = new(Conn)
+	if conn.Conn, err = dialer.Dial(network, address); err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
+// Client.DialTimeoutWithTLS acts like DialTimeoutWithTLS, but uses the client's configuration
+func (c *Client) DialTimeoutWithTLS(network, address string) (conn *Conn, err error) {
+	dialer := net.Dialer{Timeout: c.dialTimeout(), LocalAddr: c.LocalAddr}
+
+	conn = new(Conn)
+	if conn.Conn, err = tls.DialWithDialer(&dialer, network, address, c.TLSConfig); err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
```

Comment:
I see your argument now for putting the methods here. Either way is fine with me.